### DISCUSSION TITLE
`ftd.space`

### DISCUSSION THREAD

**Discussion by Arpita-Jaiswal** [fastn-stack/fastn#1144] [Ideas & RFCs]
# RFC: Adding the `ftd.space` function to `ftd`
## 1. Introduction
The `ftd.space` function is proposed to be added to the `ftd` library. This function accepts an integer as the number of space characters (" ") to add and returns this as a string. This function aims to provide a simple and efficient way to generate a string containing a specified number of space characters.

### 2.1 `ftd.space`
The `ftd.space` function is defined as follows:

```js
ftd.space = function(arg) {
    let __arg__ = {
        a: 0,
        ...arg
    };
    if (__arg__.a <= 0) {
        return '';
    }

    return ' '.repeat(__arg__.a);
}
```

Parameters
- arg (object): An object containing optional properties.
- a (number): The number of space characters to add. Default value is 0.

Return Value
- Returns a string containing the specified number of space characters.

## 3. Usage

The `ftd.space` function can be used within `ftd.text` to add space characters to the output. Here's an example of its usage:

```ftd
-- ftd.text: $ftd.space(a = 1)
```

In this example, the `ftd.space` function is used within `ftd.text` to add a single space character to the output.

**Comment by amitu** [comment]
This has moved to a [RFC PR](https://github.com/fastn-stack/fastn.com/pull/194).
